lms/static/scripts/frontend_apps/components/YouTubePicker.tsx

```diff
@@ -83,7 +83,7 @@ export default function YouTubePicker({
       <URLFormWithPreview
         onURLChange={onURLChange}
         onInput={resetCurrentURL}
-        error={error}
+        errors={error && [error]}
```

Comment:
I think pre-formatting the message could work nicely. Some logic in `YouTubePicker` that generates something like...

For a single error:

```
This video cannot be used as assignment content because it contains age-restricted content.

```

or

```
This video cannot be used as assignment content because there is no transcript available.
```

Or for multiple errors:

```
<>
This video cannot be used as assignment content because:
<ul>
  <li>it contains age-restricted content</li>
  <li>the video's owner does not allow this video to be embedded</li>
</ul>
</>
```

I could help with this part!

Point being that I think it might be more clean if `YouTubePicker` took the responsibility for formatting the error and `URLFormWithPreview` just renders what it's given. What do you think?

Reply:
Thanks!

Yeah, I like that idea. I'll close this then, as there should be no conflict in doing that in parallel if we go that way.